api/numaresourcesoperator/v1/numaresourcesoperator_types.go

```diff
@@ -114,6 +114,10 @@ type NodeGroupConfig struct {
 // NodeGroup defines group of nodes that will run resource topology exporter daemon set
 // You can choose the group of node by MachineConfigPoolSelector or by NodeSelector
 type NodeGroup struct {
+	// Name is the name used to identify this node group. Has to be unique among node groups.
+	// If not provided, is determined by the system.
+	// +optional
+	Name string `json:"name,omitempty"`
```

Comment:
let's use a string pointer so it's obvious when user set it and when not (also, "" is illegal: if user bothers to set, it should set a valid value. Using a pointer we can distinguish it)